README.md

```diff
@@ -2,6 +2,102 @@
 
+Doing migrations is the major feature of pipeline-migration-tool. It auto-discovers migrations for
+given task bundle upgrades and applies found migrations to build pipelines. The pipeline can be
+either included in a `PipelineRun` definition as `spec.pipelineSpec` or a single `Pipeline` defintion.
+
+To make it work, migrations must be attached to corresponding task bundles as OCI artifacts,
+typically with `oras attach`. Let's go deep dive a bit.
+
+* If a task bundle has a migration, it must be annotated with `dev.konflux-ci.task.has-migration: true`
+* A migration must be attached to the bundle with artifact type `text/x-shellscript` and annotated
+  with `dev.konflux-ci.task.is-migration: true`.
+* A task bundle has only one migration.
+* A migration is written as a normal Bash script. Generally, it invokes `yq` to modify the pipelines.
+
+Document [Task Migration]
+(https://github.com/konflux-ci/build-definitions/?tab=readme-ov-file#task-migration) of
+konflux-ci/build-definitions describes the migrations in detail. build-definitions provides a rich
+tool chain and CI for creation, validation and ensuring migrations are available to
+pipeline-migration-tool.
+
+## Commands
+
+Pipeline migration tool supports several sub-commands to maintain the Konflux build pipelines.
+`migrate` is the main one to do migrations. Following are a few examples of the commands usage. For
+detailed information, refer to the help messages by running each one with `-h`.
+
+### To apply migrations:
+
+```bash
+cd path/to/repo
+pipeline-migration-tool migrate -u '<upgrades>'
+```
+
+`upgrades` is a JSON string encoded from a list of mappings. Each mapping includes data for a
+single task bundle upgrade, for example:
+
+```jsonc
+[
+  {
+    "depName": "quay.io/konflux-ci/tekton-catalog/task-init",
+    "currentValue": "0.1",
+    "currentDigest": "sha256:...",
+    "newValue": "0.1",
+    "newDigest": "sha256:...",
+    "packageFile": ".tekton/component-name-pull.yaml",
+    "parentDir": ".tekton",
+    "depTypes": ["tekton-bundle"]
+  },
+  // ...
+]
+```
+
+The field names map to the [Renovate template fields]
+(https://docs.renovatebot.com/templates/#other-available-fields) directly:
+
+* `depName`: `{{depName}}`
+* `currentValue`: `{{currentValue}}`
+* `currentDigest`: `{{currentDigest}}`
+* `newValue`: `{{newValue}}`
+* `newDigest`: `{{newDigest}}`
+* `packageFile`: `{{packageFile}}`
+* `parentDir`: `{{parentDir}}`
+* `depTypes`: `{{depTypes}}`
+
+To generate the lists, using handlebars built-in `each` helper supported by Renovate.
+
+### Add a Konflux task to build pipeline
+
+Sub-command `add-task` provides rich options to add a Konflux task to build pipelines in local
+Component repositories. Let's take the task `sast-coverity-check` as an example to see a few
+command usages:
+
+* Add task with latest bundle to pipelines from inside a repository:
+
+  ```bash
+  pipeline-migration-tool add-task sast-coverity-check
```

Comment:
It's not clear where the task is inserted.
Do we support any tasks or only from predefined list? It's not clear from this doc.

Reply:
> It's not clear where the task is inserted.

The next sentence tells that.

> Do we support any tasks or only from predefined list? It's not clear from this doc.

What do you mean "any tasks" and "predefined list"?

The first paragraph mentions that this sub-command adds a Konflux task to build pipelines. This README gives general description and `add-task -h` gives more detailed information.

Reply:
> What do you mean "any tasks" and "predefined list"?

Which tasks a user can add? Can I do `pipeline-migration-tool add-task abcd1234`?

> This README gives general description and add-task -h gives more detailed information.

IMO, it would be better to have at least brief explanation here, so I don;t need to install the tool to know what it can do.

Reply:
> ### Add a Konflux task to build pipeline
> 
> Sub-command `add-task` provides rich options to add a Konflux task to build pipelines in local
> Component repositories. Let's take the task `sast-coverity-check` as an example to see a few
> command usages:

Both the heading and the paragraph mention that this is about adding a Konflux task. Isn't it enough to give enough context about what task can be added? Can you please suggest a description?